sql/create.go

```diff
@@ -142,6 +142,8 @@ func (p *planner) CreateIndex(n *parser.CreateIndex) (planNode, error) {
 		return nil, err
 	}
 
+	// Mark transaction as operating on the system DB.
+	p.txn.SetSystemDBTrigger()
```

Comment:
the setting needs to be persisted for sql transactions. eg: given the following client commands (one sql call for each):
* begin transaction
* create table
* write data to table
* commit transaction

If we don't persist the setting in the transaction proto, we'll drop it after "create table" and "write data", and never run the trigger.